codegen-v2/src/main/kotlin/club/doki7/babel/cdecl/tokenizer.kt

```diff
@@ -0,0 +1,260 @@
+package club.doki7.babel.cdecl
+
+abstract class BaseToken(
+    val value: String,
+    val line: Int,
+    val col: Int,
+) {
+    override fun toString(): String {
+        return "$line:$col:$value"
+    }
+}
+
+const val CONST = "const"
+const val VOLATILE = "volatile"
+const val TYPEDEF = "typedef"
+
+enum class TokenType {
```

Comment:
What's the point of distinguishing `TokenType` from `RawTokenType`?

Reply:
primarily for unifying `MACRO` `(` `PARAM` `)` into a single token, and extracting out possible macros